### DISCUSSION TITLE
Sterowanie pompą co/cwu

### DISCUSSION THREAD

**Discussion by bzyk1986** [krzychoooo/Miarkownik#16] [Ideas]
Witam,

Czy jest możliwość dodania do pcb miarkownika i aplikacji, funkcji sterowania pompą co/cwu. 
1. Funkcja : włączania/wyłaczania przekaźnika sterującego pompą co/cwu przy zadanej temp. odczytanej z czujnika temp.
2. Funkcja : właczenia alarmu w aplikacji, oraz sygnał dzwiękowy z miarkownika, gdy zostanie przekroczona temp. 90°C

Pozdrawiam
Symon

**Comment by krzychoooo** [comment]
Trzeba zmontować takie coś:
![Przechwytywanie1](https://user-images.githubusercontent.com/7173366/103490463-2f077200-4e1c-11eb-9098-5146dbf85b47.PNG)
Sterowanie punkt **A**
![Przechwytywanie](https://user-images.githubusercontent.com/7173366/103490476-47778c80-4e1c-11eb-9727-514598bcd84c.PNG)

Na temat softu postaram się w tygodniu coś skrobnąć.

**Reply by bzyk1986** [reply]
Witam,
I jak idą prace odnośnie softu do sterowania pompą co?

Pozdr. Szymon

**Reply by krzychoooo** [reply]
W weekend skończe :-)

**Reply by bzyk1986** [reply]
Witam krzychoooo,
Znalazłeś może trochę czasu na dokończenie softu do sterowania pompą CO ?. Bez softu samo podłączenie do pcb jest bezużyteczne.

Pzdr.

**Reply by krzychoooo** [reply]
OK. Biorę się za klepanie kodu.

**Reply by krzychoooo** [reply]
W Hardware robimy takie zmiany:
![zmiany_dla_pompy_CO_w_miarkowniku_V1](https://user-images.githubusercontent.com/7173366/105609571-7e511a80-5daa-11eb-96a6-69aead329dbe.PNG)

W rules zamieniamy na:
```
on System#Boot do
TaskValueSet 2,1,25
TaskValueSet 2,2,38
TaskValueSet 2,3,0
TaskValueSet 3,4,0
TaskValueSet 4,1,100
timerSet,2,5
Servo,1,0,25
gpio,5,1
endon 

on Termometr_1#Temperature do
  TaskValueSet 3,3,[Termometr_1#Temperature]-[piec2#oldTemp]
  if [Termometr_1#Temperature]<[piec#temp_set]
     TaskValueSet 2,4,[piec#temp_set]-[Termometr_1#Temperature]  //set error
     TaskValueSet 2,3,[piec#sumError]+[piec#error]-0.3
     TaskValueSet 2,1,[piec#error]*17+33+[piec#sumError]
  endif

  if [Termometr_1#Temperature]>[piec#temp_set] 
     TaskValueSet 2,1,5
     TaskValueSet 2,3,0
     TaskValueSet 4,1,100
  endif

  if [piec#alpha]<10
    TaskValueSet 2,1,5
  endif
  
  if [piec#alpha]>100
    TaskValueSet 2,1,100
  endif
  if [piec2#deltaTemp]<-0.2
    TaskValueSet 2,1,50      //set alpha
  endif
  if [piec2#deltaTemp]>0.2
    TaskValueSet 2,1,5      //set alpha
    TaskValueSet 2,3,-10       //set sumError
  endif

  if [piec2#stopWork]=1
    TaskValueSet 2,1,0
  endif

  if [piec#alpha]!=[piec2#oldAlpha]
    Servo,1,0,[piec#alpha]
    timerSet,1,5
  endif
  TaskValueSet 3,1,[piec#alpha]       //set piec2#oldAlpha
  TaskValueSet 3,2,[Termometr_1#Temperature]   //set piec2#oldTemp
endon

on setTemp do                        // http://<IP>/control?cmd=event,setTemp=55
  TaskValueSet 2,2,%eventvalue%
endon
on stopWorkEvent do  // zamykanie
  TaskValueSet 3,4,%eventvalue%
  if [piec2#stopWork]=0
    TaskValueSet 4,1,400
    gpio,5,1
  else
    gpio,5,0
  endif
endon

On Rules#Timer=1 do
   servo,1,0,9001
endon
On Rules#Timer=2 do
  TaskValueSet 4,1,[piec3#timeToOFF]-2
    if [piec3#timeToOFF]<0
      TaskValueSet 3,4,1   //fuel end -> stopWork
      TaskValueSet 4,1,0
      gpio,5,0
    endif
  timerSet,2,1
endon

```

**Reply by krzychoooo** [reply]
Pompa włączy się gdy:
1. start urządzenia
2. w aplikacji klikniemy przycisk "Praca"

Pompa wyłączy się gdy:
1. nastąpi koniec paliwa
2. w aplikacji klikniemy "Zamknij"

**Comment by bzyk1986** [comment]
Super,

Dzięki za szybki odzew, niedługo wezmę się za testy :)

pozdrawiam

sob., 23 sty 2021, 18:46 użytkownik krzychoooo <notifications@github.com>
napisał:

**Comment by Karolnaz** [comment]
Czy użycie przekaźnika półprzewodnikowego było by łatwiejsze i możliwe? 5V, 1,5mA

**Reply by krzychoooo** [reply]
Proszę podaj typ  przekaźnika półprzewodnikowego. W większości przypadków powinno zadziałać.

**Comment by Karolnaz** [comment]
Przepraszam, że dopiero odpisuję.
Myślałem o jakimś małym przekaźniku (obecne pompy potrafią pobierać poniżej 20W)
Znalazłem natomiast taki moduł:
https://sklep.avt.pl/modul-przekaznika-polprzewodnikowego-ssr-1-kanalowego-2a-250vac-5vdc.html
Nie jestem pewny jego sterowania

**Reply by krzychoooo** [reply]
Sygnał z miarkownika ma poziom 3,3V a ten SSR wymaga 5V. A wpięcie go zamiast cewki to znowu 12V jest wyższe od 5V. 
Trzeba by dołożyć stabilizator 5V. Szkoda że sprzedawca nie daje linku do pdf. Być może 3,3V było by ok, być może 12V też byłoby ok ????

**Reply by Karolnaz** [reply]
https://botland.com.pl/index.php?controller=attachment&id_attachment=2123

**Reply by Karolnaz** [reply]
Raczej nie przejdzie. powinno być 4-6V. Zapewnione działanie od 4V

**Reply by krzychoooo** [reply]
chyba ta wersja G3MB-202P-DC12 wpięta w kolektor (zamiast cewki)

**Comment by Plv89** [comment]
Czy mozna gdziez zakupic od ciebie taki miarkownik?
